RHIDP-6507 - modularize _Monitoring and logging with AWS (#1089)
redhat-developer/red-hat-developers-documentation-rhdh: modules/observe/proc-configuring-annotations-for-monitoring-with-amazon-prometheus-by-using-the-operator.adoc

> [id="configuring-annotations-for-monitoring-with-amazon-prometheur-by-using-the-operator_{context}"]

Author: ```suggestion
[id="configuring-annotations-for-monitoring-with-amazon-prometheus-by-using-the-operator_{context}"]
```

= Configuring annotations for monitoring with Amazon Prometheus by using the {product} Operator

> You can configure the annotations for monitoring by using the {product} Operator.

Author: ```suggestion
To enable logging to Amazon Prometheus, you can configure the required pod annotations by using the {product} Operator.
```

[id="configuring-annotations-for-monitoring-with-amazon-prometheur-by-using-the-operator_{context}"]
= Configuring annotations for monitoring with Amazon Prometheus by using the {product} Operator

You can configure the annotations for monitoring by using the {product} Operator.

.Procedure
. As an administrator of the {product} Operator, edit the default configuration to add Prometheus annotations as follows:
+
----
# Update OPERATOR_NS accordingly
$ OPERATOR_NS=rhdh-operator
$ kubectl edit configmap backstage-default-config -n "${OPERATOR_NS}"
----

> . Find the `deployment.yaml` key in the ConfigMap and add the annotations to the `spec.template.metadata.annotations` field as follows:

Author: ```suggestion
. Find the `deployment.yaml` key in the config map and add the annotations to the `spec.template.metadata.annotations` field as follows:
```

+
[source,yaml]
----
deployment.yaml: |-
  apiVersion: apps/v1
  kind: Deployment
  # --- truncated ---
  spec:
    template:
      # --- truncated ---
      metadata:
        labels:
         rhdh.redhat.com/app:  # placeholder for 'backstage-<cr-name>'
        # --- truncated ---
        annotations:
          prometheus.io/scrape: 'true'
          prometheus.io/path: '/metrics'
          prometheus.io/port: '9464'
          prometheus.io/scheme: 'http'
  # --- truncated ---
----

. Save your changes.

.Verification
To verify if the scraping works:

. Use `kubectl` to port-forward the Prometheus console to your local machine as follows:
+

> [source]

Author: ```suggestion
```

[id="configuring-annotations-for-monitoring-with-amazon-prometheur-by-using-the-operator_{context}"]
= Configuring annotations for monitoring with Amazon Prometheus by using the {product} Operator

You can configure the annotations for monitoring by using the {product} Operator.

.Procedure

> . As an administrator of the operator, edit the default configuration to add Prometheus annotations as follows:

Reviewer: Should we say "Operator" with the uppercase O?

Author: ```suggestion
. As an administrator of the {product} Operator, edit the default configuration to add Prometheus annotations as follows:
```

+

> [source]

Reviewer 2: Is this yaml, if so consider using the [source] tag to apply yaml highlighting

Author: ```suggestion
```
